src/bridge/SequencerInbox.sol

```diff
@@ -983,6 +983,27 @@ contract SequencerInbox is
     emit OwnerFunctionCalled(1);
   }
 
+  function setIsBatchPoster(
+    bytes calldata espressoMetadata
+  ) external onlyRollupOwnerOrBatchPosterManager {
+    (
+      address batchPoster,
+      bool isBatchPoster_,
+      bytes memory attestation,
+      bytes memory auxData,
+      IEspressoTEEVerifier.TeeType teeType
+    ) = abi.decode(
```

Comment:
Out of curiosity why would we want to switch to using the a single `bytes calldata` argument here as opposed to just having all of these as function arguments? This will require us to also abi encode all of the data on the batch poster side, and it might make it more clunky to all this function from cast.

Reply:
I agree with your perspective. I just wanted to reproduce a similar pattern to what we had previously but it's true.